indico/web/client/js/react/components/DateRangePicker.jsx

```diff
@@ -0,0 +1,238 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import DatePickerCalendar from 'indico/react/components/DatePickerCalendar';
+import {FinalField, validators as v} from 'indico/react/forms';
+import {Param, Translate} from 'indico/react/i18n';
+import {DateRange} from 'indico/utils/date';
+import {formatDate, ISO_FORMAT} from 'indico/utils/date_format';
+import {fromISOLocalDate} from 'indico/utils/date_parser';
+
+import 'indico/custom_elements/ind_date_picker';
+
+const INVALID = '__invalid__';
+
+export default function DateRangePicker({
+  value,
+  label,
+  format,
+  rangeStartLabel,
+  rangeEndLabel,
+  rangeStartMin,
+  rangeStartMax,
+  rangeEndMin,
+  rangeEndMax,
+  min,
+  max,
+  readOnly,
+  disabled,
+  onChange,
+}) {
+  format ??= moment.localeData().longDateFormat('L');
+
+  if (min) {
+    rangeStartMin = rangeStartMax = min;
+  }
+
+  if (max) {
+    rangeEndMin = rangeEndMax = max;
+  }
+
+  function handleChange(ev) {
+    const picker = ev.currentTarget;
+    const [start, end] = picker.dateRange;
+    const rangeStartInput = picker.querySelector('[data-range-start]');
+    const rangeEndInput = picker.querySelector('[data-range-end]');
+
+    const startInvalid = rangeStartInput.value && !start;
+    const endInvalid = rangeEndInput.value && !end;
+
+    onChange({
+      // ISO local dates
+      startDate: startInvalid ? INVALID : formatDate(ISO_FORMAT, start) || null,
+      endDate: endInvalid ? INVALID : formatDate(ISO_FORMAT, end) || null,
+    });
+  }
+
+  return (
+    <ind-date-range-picker
+      range-start={fromISOLocalDate(value.startDate)?.toDateString() || ''}
+      range-end={fromISOLocalDate(value.endDate)?.toDateString() || ''}
+      range-start-min={rangeStartMin}
+      range-start-max={rangeStartMax}
+      range-end-min={rangeEndMin}
+      range-end-max={rangeEndMax}
+      onChange={handleChange}
+    >
+      <fieldset>
+        <legend>{label}</legend>
+        <label>
+          <span>{rangeStartLabel}:</span>
+          <input
+            type="text"
+            data-range-start
+            placeholder={format}
+            readOnly={readOnly}
+            disabled={disabled}
+          />
+        </label>
+        <span className="arrow" />
+        <label>
+          <span>{rangeEndLabel}:</span>
+          <input
+            type="text"
+            data-range-end
+            placeholder={format}
+            readOnly={readOnly}
+            disabled={disabled}
+          />
+        </label>
+        <button type="button" data-calendar-trigger>
```

Comment:
this one is still the case (both for datepicker and daterangepicker)

Reply:
Hmmm, I tried reproducing this on Chrome and Firefox but I couldn't. Can you give me a step-by-step of what exactly you're doing and also tell me what browser you're using?

Reply:
I'm always using Firefox (latest beta release).

In Room Booking it no longer disables the trigger button to open the date picker while it's open. In the registration form it still works fine though... very strange!

![image](https://github.com/user-attachments/assets/7170287f-0ef1-4207-a713-259c9841cdcb)

Reply:
Hmmm. That sounds like CSS from somewhere else interfering. Cause I literally just use `pointer-events` to disable the button in CSS.

Reply:
Or hmmm. No, it actually has a disabled attr as well.

Reply:
When I click it with devtools open the disabled attribute goes away when pressing the button, and then releasing it again then triggers the usual click that reopens the picker.

Reply:
So here's what's going on. In the reg form, the field is located inside a SUI field thing, it that's where it gets its pointer-events property (and 0.45 opacity). In the Room Booking it does not. I'm gonna duplicate the style from SUI in the custom element CSS module tomorrow so that it does the same thing regardless of where it's used.